File: test/src/main/java/com/shipping/service/FileService.java
package com.shipping.service;

import com.shipping.domain.sys.File;
import com.shipping.repository.sys.FileRepository;
import org.spin.web.FileOperator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.UUID;

/**
 * 文件服务
 * <p>Created by xuweinan on 2017/4/26.</p>
 *
 * @author xuweinan
 */
@Service
public class FileService {

    @Autowired
    private FileRepository fileDao;

    @Transactional
    public File saveFile(FileOperator.UploadResult uploadResult) {
        File file = new File();
        file.setGuid(UUID.randomUUID().toString());
        file.setOriginName(uploadResult.getOriginName());
        file.setFileName(uploadResult.getStoreName().substring(uploadResult.getStoreName().lastIndexOf('/') + 1));
        file.setFilePath(uploadResult.getStoreName());
        file.setExtension(uploadResult.getExtention());
        file.setSize(uploadResult.getSize());
        return fileDao.save(file);
    }

    @Transactional
    public List<File> saveFiles(Collection<FileOperator.UploadResult> uploadResults) {
        List<File> result = new ArrayList<>();
        for (FileOperator.UploadResult uploadResult : uploadResults) {
            File file = new File();
            file.setGuid(UUID.randomUUID().toString());
            file.setOriginName(uploadResult.getOriginName());
            file.setFileName(uploadResult.getStoreName().substring(uploadResult.getStoreName().lastIndexOf('/') + 1));
            file.setFilePath(uploadResult.getStoreName());
            file.setExtension(uploadResult.getExtention());
            file.setSize(uploadResult.getSize());
            result.add(fileDao.save(file));
        }
        return result;
    }
}

File: test/src/main/kotlin/com/shipping/service/FileService.kt
package com.shipping.service

import com.shipping.domain.sys.File
import org.spin.data.extend.RepositoryContext
import org.spin.web.FileOperator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*

/**
 * 文件服务
 *
 * Created by xuweinan on 2017/4/26.
 *
 * @author xuweinan
 */
@Service
class FileService {

    @Autowired
    private lateinit var repoCtx: RepositoryContext

    @Transactional
    fun saveFile(uploadResult: FileOperator.UploadResult): File =
        repoCtx.save(File(guid = UUID.randomUUID().toString(),
            originName = uploadResult.originName,
            fileName = uploadResult.storeName.substring(uploadResult.storeName.lastIndexOf('/') + 1),
            filePath = uploadResult.storeName,
            extension = uploadResult.extention,
            size = uploadResult.size
        ))

    @Transactional
    fun saveFiles(uploadResults: Collection<FileOperator.UploadResult>): List<File> =
        uploadResults.map {
           saveFile(it)
        }

}
